index.js

```diff
@@ -1,18 +1,90 @@
 const readline = require('readline');
 const rl = readline.createInterface(process.stdin, process.stdout);
+// Global Variables Assigned Here
+let answer
 
+// Promise Function
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
-
-start();
-
+// Return Random Number Funciton
+function randomNum(min, max) {
+  return Math.floor(Math.random() * (max - min + 1) + min)
+}
+// Return Mid Value Function (whole number)
+function midValue(min, max) {
+  return Math.floor((min + max) / 2)
+}
+// Start Game Function
+start()
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  console.log('Welcome to my Guessing Game!\n' + '*Games alternate after intital selection*\n')
+  let gameChoice = await ask('To Start Please Choose Either Computer Guesses or Human Guesses\n')
+  gameChoice = gameChoice.trim().toLowerCase()
+  if (gameChoice == 'computer guesses') {
+    computerGuesses()
+  }
+  if (gameChoice == 'human guesses') {
+    humanGuesses()
+  }
 }
+// Computer Guess Game Function
+async function computerGuesses() {
+  console.log('\nYou have now entered Mode: Computer Guesses\n' + 'To play this game please choose a numerical range for the game(ex. 1 to 100)')
+  let userMinRange = parseInt(await ask('What is your desired low end range?\n'));
+  let userMaxRange = parseInt(await ask('What is your desired top end range?\n'));
+  console.log('You entered: ' + userMinRange + ' to ' + userMaxRange)
+  let computerGenNum = randomNum(userMinRange, userMaxRange)
```

Comment:
It's a small thing, but to fully minimize the computer's number of guesses,  the first guess should be **smart** too, not random. This would effectively remove the need for your `randomNum` function altogether.  The first guess sets the trend for all subsequent guesses and their ranges, so the need to start this process with the initial ranges is important in maximizing efficiency.